rewrite-maven/src/main/java/org/openrewrite/maven/AddOrUpdateDependencyManagement.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import org.openrewrite.Formatting;
+import org.openrewrite.Validated;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.maven.tree.MavenModel;
+import org.openrewrite.refactor.Formatter;
+import org.openrewrite.xml.XmlParser;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+import static java.util.Collections.emptyList;
+import static org.openrewrite.Formatting.format;
+import static org.openrewrite.Tree.randomId;
+import static org.openrewrite.Validated.required;
+
+/**
+ * Adds a dependency if there is no dependency matching <code>groupId</code> and <code>artifactId</code>.
+ * A matching dependency with a different version or scope does NOT have its version or scope updated.
+ * Use {@link ChangeDependencyVersion} or {@link UpgradeDependencyVersion} in the case of a different version.
+ * Use {@link ChangeDependencyScope} in the case of a different scope.
+ * <p>
+ * Places a new dependency as physically "near" to a group of similar dependencies as possible.
+ */
+public class AddOrUpdateDependencyManagement extends MavenRefactorVisitor {
+    private String groupId;
+    private String artifactId;
+    private String description;
+
+    public String getDescription() {
+        return description;
+    }
+
+    @Nullable
+    private String version;
+
+    @Nullable
+    private String scope;
+
+    public void setGroupId(String groupId) {
+        this.groupId = groupId;
+    }
+
+    public void setArtifactId(String artifactId) {
+        this.artifactId = artifactId;
+    }
+
+    public void setVersion(@Nullable String version) {
+        this.version = version;
+    }
+
+    public void setScope(@Nullable String scope) {
+        this.scope = scope;
+    }
+
+    @Override
+    public Validated validate() {
+        return required("groupId", groupId)
+                .and(required("artifactId", artifactId));
+    }
+
+    @Override
+    public boolean isIdempotent() {
+        return false;
+    }
```

Comment:
`super.isIdempotent()` is going to return `false` by default. Not necessary to override `isIdempotent()` in this case